lib/Doctrine/Parser/sfYaml/sfYamlInline.php

```diff
@@ -180,6 +180,8 @@ function($v, $w) {
    */
   static public function parseScalar($scalar, $delimiters = null, $stringDelimiters = array('"', "'"), &$i = 0, $evaluate = true)
   {
+    $scalar = (string) $scalar;
```

Comment:
The character array access below causes the static analyzers to complain because you can't do that on `bool`, `int` or `float` types, just strings.